browser/ui/views/toolbar/brave_toolbar_view_browsertest.cc

```diff
@@ -112,9 +141,7 @@ IN_PROC_BROWSER_TEST_F(BraveToolbarViewTest,
 
   // Open the new profile
   EXPECT_EQ(1U, BrowserList::GetInstance()->size());
-  content::WindowedNotificationObserver browser_creation_observer(
-      chrome::NOTIFICATION_BROWSER_OPENED,
-      content::NotificationService::AllSources());
+  BrowserAddedObserver browser_creation_observer;
```

Comment:
maybe this is `WaitForBrowserToOpen()` from `ui_test_utils.h` ?

Reply:
Good point, I was not aware of that utility, thanks!

Fixed in https://github.com/brave/brave-core/commit/2b7f27ffe7b576f1d23f438442393854c44d5117

Reply:
cc: @mariospr WDYT? It looks the same, but I wonder about the timing: in our code we set up the observer ahead of the code that would open a new browser, whereas `WaitForBrowserToOpen` is called after the code, which makes me wonder if it might miss the event.

Reply:
Sorry, didn't see your comments when I posted @mariospr

Reply:
Locally on Windows: 
<pre>
8 tests failed:
    BraveAppMenuBrowserTest.MenuOrderTest (../../brave/browser/ui/toolbar/brave_app_menu_model_browsertest.cc:67)
    BraveBrowserCommandControllerTest.BraveCommandsEnableTest (../../brave/browser/ui/brave_browser_command_controller_browsertest.cc:37)
    BraveToolbarViewTest.AvatarButtonIsShownGuestProfile (../../brave/browser/ui/views/toolbar/brave_toolbar_view_browsertest.cc:67)
    BraveToolbarViewTest.AvatarButtonIsShownMultipleProfiles (../../brave/browser/ui/views/toolbar/brave_toolbar_view_browsertest.cc:87)
    OnionLocationNavigationThrottleBrowserTest.OnionDomain (../../brave/browser/tor/onion_location_navigation_throttle_browsertest.cc:143)
    OnionLocationNavigationThrottleBrowserTest.OnionDomain_AutoOnionRedirect (../../brave/browser/tor/onion_location_navigation_throttle_browsertest.cc:165)
    OnionLocationNavigationThrottleBrowserTest.OnionLocationHeader (../../brave/browser/tor/onion_location_navigation_throttle_browsertest.cc:123)
    OnionLocationNavigationThrottleBrowserTest.OnionLocationHeader_AutoOnionRedirect (../../brave/browser/tor/onion_location_navigation_throttle_browsertest.cc:209)
</pre>

Reply:
Pushed a fix (2f371c52fef419fb67ad9ddbbad7ec69eca0605e) that goes half way: instantiates a ui_test_utils::BrowserChangeObserver instance early and then does then wait when needed.

Reply:
@mkarolin Oops! I have to apologize because I slipped on this: I was in a bit of a rush yesterday with other things and I made the mistake of addressing @iefremov 's review comments without paying attention to the result of running the tests (just making sure it compiled), otherwise I would have noticed the problem you mentioned and wouldn't have submitted my patch as it was...

As for your patch at https://github.com/brave/brave-core/commit/2f371c52fef419fb67ad9ddbbad7ec69eca0605e , I've checked it and I think it makes more sense, and I particularly liked how you removed those loops and simply used the return value of `browser_creation_observer.Wait()` instead.

Again, sorry and thanks for fixing up my mess.

Reply:
One more fix to solve 2 test failures only happening on linux: https://github.com/brave/brave-core/commit/278beddde5a72a0f33e8ff46e70530457303782e